lib/bolt/module_installer.rb

```diff
@@ -114,7 +114,16 @@ def install(modules, path, moduledir, force: false, resolve: true)
           puppetfile.resolve
 
           @outputter.print_message "Writing Puppetfile at #{path}"
-          puppetfile.write(path)
+          # We get here either through 'bolt module install' which uses the
+          # managed modulepath (which isn't configurable) or through bolt
+          # project init --modules, which uses the default modulepath. This
+          # should be safe to assume that if `.modules/` is the moduledir the
+          # user is using the new workflow
+          if moduledir.basename == '.modules'
```

Comment:
```suggestion
          if File.basename(moduledir) == '.modules'
```

`moduledir` (and all other parts of the modulepath) are strings.

Reply:
... ?
```
lucy ~/githubs/bolt/test (GH-2234) $ be bolt project init --modules puppetlabs-apache
Resolving module dependencies, this may take a moment
Writing Puppetfile at /home/lucy/githubs/bolt/test/Puppetfile

Frame number: 0/23

From: /home/lucy/githubs/bolt/lib/bolt/module_installer.rb:123 Bolt::ModuleInstaller#install:

    118:           # managed modulepath (which isn't configurable) or through bolt
    119:           # project init --modules, which uses the default modulepath. This
    120:           # should be safe to assume that if `.modules/` is the moduledir the
    121:           # user is using the new workflow
    122:           require 'pry'; binding.pry
 => 123:           if moduledir.basename == '.modules'
    124:             puppetfile.write(path, moduledir)
    125:           else
    126:             puppetfile.write(path)
    127:           end
    128:         end

[1] pry(#<Bolt::ModuleInstaller>)> moduledir.class
=> Pathname
```

Reply:
Oh UGH. It's a pathname if it gets there through `bolt project init --modules`, but a string if it gets there through `bolt module install` since apparently the moduledir on the project is the *only* path that's a string?